crates/hir-def/src/nameres/collector.rs

```diff
@@ -1613,7 +1615,10 @@ impl ModCollector<'_, '_> {
                 ModItem::Struct(id) => {
                     let it = &self.item_tree[id];
 
-                    let vis = resolve_vis(def_map, &self.item_tree[it.visibility]);
+                    self.process_fields(&it.fields);
```

Comment:
I've moved the collection to the queries but the diff is still large due to https://github.com/rust-lang/rust-analyzer/pull/13189#discussion_r962349641 (because of `From` impls for lots of types)